File: test/c++11/intel/rdmsr.cpp
#include <pal/instruction/rdmsr.h>
#include <pal/msr/ia32_feature_control.h>
#include <stdio.h>

void test_rdmsr_compile(void)
{
    uint32_t address = pal::ia32_feature_control::address;
    uint64_t value = pal::execute_rdmsr(address);
    pal::ia32_feature_control::print(printf, value);
}

File: test/rust/intel/src/rdmsr.rs
use pal;

pub unsafe fn test_rdmsr_compile()
{
    let address = pal::msr::ia32_feature_control::ADDRESS;
    let value = pal::instruction::execute_rdmsr(address);

    #[cfg(feature = "pal/enable_printers")]
    pal::msr::ia32_feature_control::print_from_value(value);

    #[cfg(not(feature = "pal/enable_printers"))]
    let _ = value;
}
